haystack/document_stores/elasticsearch.py

```diff
@@ -180,7 +180,7 @@ def __init__(
         self.scroll = scroll
         self.skip_missing_embeddings: bool = skip_missing_embeddings
         if similarity in ["cosine", "dot_product", "l2"]:
-            self.similarity = similarity
+            self.similarity: str = similarity
         else:
             raise Exception(
```

Comment:
I know that this is not the scope of this PR, but might be a good opportunity to change this general `Exception` to a `ValueError` or one of our `HaystackError`s.